### DISCUSSION TITLE
how to open database from exists stream ?

### DISCUSSION THREAD

**Discussion by xibaoning** [Giorgi/DuckDB.NET#140] [General]
var conn=new DuckDBConnection(stream) ??

**Comment by Giorgi** [comment]
You can't open from stream, why would you want to do that?

**Comment by Cricle** [comment]
[The duckdb api limit](https://duckdb.org/docs/api/c/connect#api-reference)

**Comment by mldisibio** [comment]
Actually, it is a [feature of sqlite](https://sqlite.org/c3ref/serialize.html) and not an entirely crazy ask.
That said, it is not part of the duckdb api, and therefore a client library wrapping the native api (IMHO) probably shouldn't try to add features not in the api itself.

It's a corner use-case. Say I download a sqlite file stored on a remote file server, knowing it fits in memory, and want to run a query without saving the file to disk. In particular, if the query is served by a web api, but the sqlite file is remote, perhaps compressed and needs to be retrieved first, but is discarded after serving the query.

**Comment by xibaoning** [comment]
For example, green software can embed data as resources within the executable. If the data can be read directly from the stream during runtime, there would be no need to extract the executable to a physical file. Perhaps you could try reaching out to the official team and see if they can assist with this.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年11月19日(星期二) 凌晨3:34
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Giorgi/DuckDB.NET] how to open database from exists stream ? (Discussion #140)














 
Actually, it is a feature of sqlite and not an entirely crazy ask.
 That said, it is not part of the duckdb api, and therefore a client library wrapping the native api (IMHO) probably shouldn't try to add features not in the api itself.
 
It's a corner use-case. Say I download a sqlite file stored on a remote file server, knowing it fits in memory, and want to run a query without saving the file to disk. In particular, if the query is served by a web api, but the sqlite file is remote, perhaps compressed and needs to be retrieved first, but is discarded after serving the query.

**Comment by Giorgi** [comment]
If DuckDB adds an API for it I will consider supporting it. Until then this isn't going to be implemented.
